### ISSUE TITLE
[2.3.56.0]Gcode preview window move part of a layer

### ISSUE THREAD

**Issue by ldiegos** [supermerill/SuperSlicer#1169]
**Describe the bug**
The version 2.3.56.0/2.3.55 is moving part of a layer to a random place and this is written in the gcode. At least on my geeetech A10M configuration, dual color printer.

**To Reproduce**
1. Install version 2.3.56.0
2. Add a doubled stl for a dual color printer.
3. See Gcode preview

**>> Project File <<**
*Please* upload a ZIP archive containing the project file used when the problem arise. Please export it just before the problem occur. Even if did nothing and there is no object, export it as it contains your current configuration.

**Expected behavior**
Not to move the layers, and maintain the stl aspect.

**Screenshots**
![image](https://user-images.githubusercontent.com/29803600/119254741-68b4fb80-bbb8-11eb-86c0-922a87ea52ac.png)

Where it should be.
![image](https://user-images.githubusercontent.com/29803600/119254751-79657180-bbb8-11eb-8394-6a3421ad4f9a.png)

**Desktop (please complete the following information):**
 - OS: Windows 10 pro
 - Version: 2.3.56.0

**Additional context**

**Comment by supermerill** [comment]
Can't reproduce
![image](https://user-images.githubusercontent.com/6536403/119404477-271f7000-bce0-11eb-8b9d-8f895f94b41c.png)

Can i have your project file?

**Comment by ldiegos** [comment]
Hi supermerill

Here are two project files.

The shift layer is on top of the model:
[dualTest3b.zip](https://github.com/supermerill/SuperSlicer/files/6540324/dualTest3b.zip)

The shift layer is in the back of the model:
[CaliCatDual.zip](https://github.com/supermerill/SuperSlicer/files/6540335/CaliCatDual.zip)

**Comment by supermerill** [comment]
what's going on:
it see your `G1 Z3` and so it (wrongly) erase the next `G1 Z3` because the cleaner see this line as a duplicate.

I won't fix it as this part will go into a full rewrite, so may be fixed after the rewrite.

For now, you can change your `G1 Z3` by `G1 Z3.001`,

**Comment by ldiegos** [comment]
Good morning supermerill!

Really the ToolChange G-code script made that? Let me try again.

Thank you so much.

**Comment by ldiegos** [comment]
Perfect!
![image](https://user-images.githubusercontent.com/29803600/119607067-ffb9c780-bdf3-11eb-86d6-2360a20ec415.png)

Perfect!
![image](https://user-images.githubusercontent.com/29803600/119607150-14965b00-bdf4-11eb-9f36-7d497953cb92.png)

will you close the ticket? i will have in mind not to use common layer values and add always a .001 to the amount.
